internal/tunnel/client/reconnecting.go

```diff
@@ -284,33 +363,79 @@ func (s *reconnectingSession) connect(acceptErr error) error {
 		}
 
 		// dial the tunnel server
-		raw, err := s.dialer()
+		raw, err := s.dialer(connSession.legNumber)
 		if err != nil {
 			failTemp(err, raw)
 			continue
 		}
 
 		// successfully reconnected
-		s.swapper.set(raw)
+		connSession.swapper.set(raw)
 
 		// callback for authentication
-		if err := s.cb(s); err != nil {
+		desiredLegs, err := s.cb(s, raw, connSession.legNumber)
+		if err != nil {
 			failTemp(err, raw)
 			continue
 		}
 
+		// check if more sessions need to be established
+		sendStateChange := true
+		if desiredLegs > len(s.sessions) {
+			// set up the next connection
+			s.createTunnelClientSession(s)
+			// not done with initial setup yet
+			sendStateChange = false
+		}
```

Comment:
Can we go ahead and make this

```suggestion
		for i := len(sessions); i < desiredLegs; i++ {
			// set up the next connection
			s.createTunnelClientSession(s)
			// not done with initial setup yet
			sendStateChange = false
		}
```

for if/when we support >2 legs?

err, wait. If `createTunnelClientSession` calls `connect`, then I guess this block will get hit each time a new leg is opened, which will drive the creation of the rest of the legs. So no loop needed?

Reply:
aye, the intent was to chain the connects along, one finishing would start the next